scaling_performance/network_optimization.adoc

```diff
@@ -67,6 +70,57 @@ xref:../architecture/core_concepts/routes.adoc#router-sharding[Router sharding]
 should be used to serve more routes towards applications and help horizontally
 scale the routing tier.
 
+[[scaling-performance-optimizing-router-haproxy]]
+=== Performance Optimizations
+
+[[scaling-performance-optimizing-router-haproxy-maxconn]]
+==== Setting the Maximum Number of Connections
+
+One of the most important tunable parameters for HAProxy is the 
+`maxconn` parameter which sets the maximum per-process number 
+of concurrent connections to a given number.  In {product-title} 
+HAProxy router this parameter is adjustable by editing 
+the 
+`xref:../install_config/router/default_haproxy_router.adoc#concurrent-connections[ROUTER_MAX_CONNECTIONS]`
+environment variable in the router's
+deployment configuration.
+
+
+[[scaling-performance-optimizing-router-haproxy-cpu-affinity]]
+==== CPU and Interrupt Affinity
+
+While it is entirely possible to run HAProxy with multiple worker
+processes using the `nbproc` parameter, in {product-title} HAProxy
+router runs as a single process.  {product-title} HAProxy router
+will typically perform better on a system with a fewer but high
+frequency cores, rather than on an SMP system with a high number of
+lower frequency cores.
+
+Pinning HAProxy process to one CPU core and the network interrupts to another,
+while all sharing the same L3 cache, tends to increase network performance
+because in practice the amount of work for HAProxy and the network stack are
+quite close.  CPU pinning is performed either by `taskset` or using HAProxy's
+`cpu-map` parameter.  This directive takes two arguments: the process id and 
+the cpu core id.  For example, to pin HAProxy process 1 onto CPU core 0, add
+the following line to the global section of HAProxy's configuration file:
+
+    cpu-map 1 0
+
+
+[[scaling-performance-optimizing-router-haproxy-bufsize]]
+==== Impacts of Buffer Increases
+
+The {product-title} HAProxy router request buffer configuration
+limits the size of headers in incoming requests and responses from
+applications.  The HAProxy parameter `tune.bufsize` can be increased
+to allow processing of larger headers and allow applications with
+very large cookies to work.  This however affects the total memory
```

Comment:
I can do that, but didn't we want to avoid any references to specific public cloud providers in the OpenShift docs?